Penteract/Assets/Scripts/AttackDroneBehavior.cpp

```diff
@@ -0,0 +1,111 @@
+#include "AttackDroneBehavior.h"
+
+#include "GameplaySystems.h"
+#include "Components/ComponentTransform.h"
+#include "Components/ComponentParticleSystem.h"
+#include "RandomNumberGenerator.h"
+
+EXPOSE_MEMBERS(AttackDroneBehavior) {
+    MEMBER(MemberType::GAME_OBJECT_UID, dronesControllerUID),
+    MEMBER(MemberType::PREFAB_RESOURCE_UID, projectilePrefabUID),
+    MEMBER(MemberType::FLOAT, droneSpeed),
+    MEMBER(MemberType::FLOAT, droneRotationSpeed),
+    MEMBER(MemberType::BOOL, droneMustHover),
+    MEMBER(MemberType::FLOAT, droneHoverAmplitude),
+    MEMBER(MemberType::BOOL, droneMustRecoil),
+    MEMBER(MemberType::FLOAT, droneRecoilOffset),
+    MEMBER(MemberType::FLOAT, droneRecoilTime),
+};
+
+GENERATE_BODY_IMPL(AttackDroneBehavior);
+
+void AttackDroneBehavior::Start() {
+    transform = GetOwner().GetComponent<ComponentTransform>();
+
+    dronesController = GameplaySystems::GetGameObject(dronesControllerUID);
+    if (dronesController) {
+        dronesControllerTransform = dronesController->GetComponent<ComponentTransform>();
+    }
+
+    hoverCurrentTime = RandomNumberGenerator::GenerateFloat(-1.5708, 1.5708);
+}
+
+void AttackDroneBehavior::Update() {
+    if (!transform || !dronesControllerTransform) return;
+
+    transform->SetGlobalPosition(float3::Lerp(transform->GetGlobalPosition(), dronesControllerTransform->GetGlobalPosition() + positionOffset + GetHoverOffset() + GetRecoilOffset(), Time::GetDeltaTime() * droneSpeed));
+    
+    if (mustForceRotation) {
+        float3 direction = (transform->GetGlobalPosition() - dronesControllerTransform->GetGlobalPosition()).Normalized();
+        float3 right = Cross(float3(0, 1, 0), direction);
+        direction = Cross(right, float3(0, 1, 0));
```

Comment:
Your method is the correct and it's general. But I don't know if it would be quicker to take the distance between positions (without normalizing),  set the y component to 0, and then normalize. Because you only want a direction that is parallel to the floor.